tokio/src/io/uring/unlink.rs

```diff
@@ -0,0 +1,57 @@
+use crate::io::uring::utils::cstr;
+use crate::runtime::driver::op::{CancelData, Cancellable, Completable, CqeResult, Op};
+use io_uring::{opcode, types};
+use std::ffi::CString;
+use std::io;
+use std::io::Error;
+use std::path::Path;
+
+#[derive(Debug)]
+pub(crate) struct Unlink {
+    /// This field will be read by the kernel during the operation, so we
+    /// need to ensure it is valid for the entire duration of the operation.
+    #[allow(dead_code)]
+    path: CString,
+}
+
+impl Completable for Unlink {
+    type Output = io::Result<()>;
+
+    fn complete(self, cqe: CqeResult) -> Self::Output {
+        cqe.result.map(drop)
+    }
+
+    fn complete_with_error(self, error: Error) -> Self::Output {
+        Err(error)
+    }
+}
+
+impl Cancellable for Unlink {
+    fn cancel(self) -> CancelData {
+        CancelData::CString(self.path)
+    }
+}
+
+impl Op<Unlink> {
+    /// Submit a request to open a file.
+    pub(crate) fn unlink(path: &Path, directory: bool) -> io::Result<Op<Unlink>> {
```

Comment:
We can have `remove_dir` and `remove_file` and you can ask for `flags` in the argument, its not apparent what the directory `bool` is going to do